packages/vulcan-lib/lib/modules/reactive-state.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * Simple state management based on Apollo Client reactive variables exposed as React HoCs and Hooks.
+ * Use it to store session data that survives re-renders and router transitions, unlike component state.
+ * Register multiple state objects with optional SimpleSchemas for cleaning and validation.
+ * @see {@link https://www.apollographql.com/docs/react/local-state/reactive-variables/}
+ *
+ * @module reactive-state
+ */
+
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {createSchema} from './schema_utils';
+import {makeVar} from '@apollo/client';
+import update from 'immutability-helper';
+// eslint-disable-next-line no-unused-vars
+import SimpleSchema from 'simpl-schema';
+import _forOwn from 'lodash/forOwn';
+
+
+const reactiveStates = {};
+
+
+/**
+ * An object for storing global state based on Apollo Client reactive variables
+ * @typedef {Object} ReactiveState
+ * @property {string} stateKey - The name/id/key of the state
+ * @property {SimpleSchema} [schema] - Optional schema
+ * @property {Object} [defaultValue] - Optional default value
+ * @property {function} reactiveVar - The reactive variable
+ */
+
+/**
+ * Create a new reactive state object
+ * @param {string} stateKey The name/id/key for the new reactive state
+ * @param {Object} [schema] Optional schema definition object that will be converted to `SimpleSchema` using `createSchema()`
+ * @param {Object} [defaultValue] Optional default value; alternatively you can define `defaultValue`s in the schema
+ * @param {boolean} [skipDuplicate] If you try to create a reactive state with a key that's already used, an exception will
+ * be thrown; use this option to prevent the exception and use the existing state without changing it
+ * @returns {ReactiveState} Returns the newly created state object
+ * @throws Will throw an error if there is already a reactive state object with the given key - unless `skipDuplicates` is `true`
+ */
+export const createReactiveState = ({stateKey, schema, defaultValue, skipDuplicate}) => {
+  if (reactiveStates[stateKey]) {
+    if (skipDuplicate) return;
+    throw new Error(`There is already a reactive state object named ${stateKey}`);
+  }
+
+  if (schema) {
+    schema = createSchema(schema);
+    defaultValue = cleanReactiveStateValue(defaultValue || {}, schema);
+  }
+
+  reactiveStates[stateKey] = {
```

Comment:
I am not fond of this register pattern anymore and would tend to deprecate them in existing application, even though those patterns are common in Vulcan. 
Is there a good reason to store all the vars in a global object? If they need to carry metadata with the object, maybe we can add those data directly in the created object? Like `const cartItemsVar = makeVar([]); carItemsVar._vulcan = { schema }; return carItemsVar;`.
The point being that you avoid to use a global register, which in turns avoid any risk of conflict and forces end user to clarify their dependencies with actual ES6 `import/export`. 

That's just a proposal though, if you feel good using it as is then I am fine with this implementation.

Reply:
Really I did it this way to follow the Vulcan pattern. I will try it without the global state and see how that feels.